docs/reference/modules/coordination.asciidoc

```diff
@@ -0,0 +1,499 @@
+POST /_nodes/node_name/retire
+# Retire node and wait for its removal up to one minute
+POST /_nodes/node_name/retire?timeout=1m
+--------------------------------------------------
+// CONSOLE
+
+The node to retire is specified using <<cluster-nodes,node filters>> in place
+of `node_name` here. If a call to the retirement API fails then the call can
+safely be retried. However if a retirement fails then it's possible the node
+cannot be removed from the configuration due to the
+`cluster.master_nodes_failure_tolerance` setting, so verify that this is set
+correctly first. A successful response guarantees that the node has been
+removed from the configuration and will not be reinstated.
+
+A node (or nodes) can be brought back out of retirement using the `unretire`
+API:
+
+[source,js]
+--------------------------------------------------
+POST /_nodes/node_name/unretire
+--------------------------------------------------
+// CONSOLE
+
+The node (or nodes) to reinstate are specified using <<cluster-nodes,node
+filters>> in place of `node_name` here. After being brought back out of
+retirement they may not immediately be added to the configuration.
+
+The current set of retired nodes is stored in the cluster state and can be
+inspected as follows:
+
+[source,js]
+--------------------------------------------------
+GET /_cluster/state?filter_path=TODO
+--------------------------------------------------
+// CONSOLE
+
+[float]
+=== Cluster bootstrapping
+
+There is a risk when starting up a brand-new cluster is that you accidentally
+form two separate clusters instead of one. This could lead to data loss: you
+might start using both clusters before noticing that anything had gone wrong,
+and it will then be impossible to merge them together later.
+
+NOTE: To illustrate how this could happen, imagine starting up a three-node
+cluster in which each node knows that it is going to be part of a three-node
+cluster. A majority of three nodes is two, so normally the first two nodes to
+discover each other will form a cluster and the third node will join them a
+short time later. However, imagine that four nodes were accidentally started
+instead of three: in this case there are enough nodes to form two separate
+clusters. Of course if each node is started manually then it's unlikely that
+too many nodes are started, but it's certainly possible to get into this
+situation if using a more automated orchestrator, particularly if a network
+partition happens at the wrong time.
+
+We avoid this by requiring a separate _cluster bootstrap_ process to take place
+on every brand-new cluster. This is only required the very first time the whole
+cluster starts up: new nodes joining an established cluster can safely obtain
+all the information they need from the elected master, and nodes that have
+previously been part of a cluster will have stored to disk all the information
+required when restarting.
+
+The simplest way to bootstrap a cluster is to use the
```

Comment:
This tool is now too magical imo. I suggest to either first explain the API and then introduce the tool (explaining what it does using api calls) or have a description of what the tool does here (ala the bootstrapping process requires two api calls described below. As a convenience we offer a command line tool that will interact with the given node urls to bootstrap the cluster).

PS - note that all of this requires changing current http semantics. Right now we wait up to 30s (by default) for the node to join a cluster before opening up the http layer.

Reply:
Ok, I reversed the order of the bootstrapping section.

And yes, we will need to start listening for HTTP earlier. I have started to discuss this with the clients team.

Reply:
Thanks. I suspect the client teams should not change anything. Our current logic is nothing more than a best effort. The same holds for the waiting for the initial cluster state to arrive.